pkg/tfgen/convert_cli.go

```diff
@@ -0,0 +1,467 @@
+	n := 0
+	for hcl := range cc.hcls {
+		fileName := fmt.Sprintf("e%d", n)
+		examples[fileName] = hcl
+		n++
+	}
+	result, err := cc.convertViaPulumiCLI(examples, []struct {
+		name string
+		info tfbridge.ProviderInfo
+	}{
+		{
+			name: cc.packageName,
+			info: cc.info,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	for fileName, hcl := range examples {
+		r := result[fileName]
+		cc.pcls[hcl] = translatedExample{
+			PCL:         r.PCL,
+			Diagnostics: cc.removeFileName(r.Diagnostics),
+		}
+	}
+	return nil
+}
+
+// Calls pulumi convert to bulk-convert examples.
+//
+// To facilitate high-throughput conversion, an `examples.json` protocol is employed to convert
+// examples in batches. See pulumi/pulumi-converter-terraform#29 for where the support is
+// introduced.
+//
+// Source examples are passed in as a map from ID to raw TF code.
+//
+// This may need to be coarse-grain parallelized to speed up larger providers at the cost of more
+// memory, for example run 4 instances of `pulumi convert` on 25% of examples each.
+func (*cliConverter) convertViaPulumiCLI(
+	examples map[string]string,
+	mappings []struct {
+		name string
+		info tfbridge.ProviderInfo
+	},
+) (
+	output map[string]translatedExample,
+	finalError error,
+) {
+	outDir, err := os.MkdirTemp("", "bridge-examples-output")
+	if err != nil {
+		return nil, fmt.Errorf("convertViaPulumiCLI: failed to create a temp dir "+
+			" bridge-examples-output: %w", err)
+	}
+	defer func() {
+		if err := os.RemoveAll(outDir); err != nil {
+			if finalError == nil {
+				finalError = fmt.Errorf("convertViaPulumiCLI: failed to clean up "+
+					"temp bridge-examples-output dir: %w", err)
+			}
+		}
+	}()
+
+	examplesJSON, err := os.CreateTemp("", "bridge-examples.json")
+	if err != nil {
+		return nil, fmt.Errorf("convertViaPulumiCLI: failed to create a temp "+
+			" bridge-examples.json file: %w", err)
+	}
+	defer func() {
+		if err := os.Remove(examplesJSON.Name()); err != nil {
+			if finalError == nil {
+				finalError = fmt.Errorf("convertViaPulumiCLI: failed to clean up "+
+					"temp bridge-examples.json file: %w", err)
+			}
+		}
+	}()
+
+	// Write example to bridge-examples.json.
+	examplesBytes, err := json.Marshal(examples)
+	if err != nil {
+		return nil, fmt.Errorf("convertViaPulumiCLI: failed to marshal examples "+
+			"to JSON: %w", err)
+	}
+	if err := os.WriteFile(examplesJSON.Name(), examplesBytes, 0600); err != nil {
+		return nil, fmt.Errorf("convertViaPulumiCLI: failed to write a temp "+
+			"bridge-examples.json file: %w", err)
+	}
+
+	mappingsDir := filepath.Join(outDir, "mappings")
+
+	mappingsFile := func(name string) string {
+		return filepath.Join(mappingsDir, fmt.Sprintf("%s.json", name))
+	}
+
+	// Write out mappings files if necessary.
+	for i, m := range mappings {
+		if i == 0 {
+			if err := os.MkdirAll(mappingsDir, 0755); err != nil {
+				return nil, fmt.Errorf("convertViaPulumiCLI: failed to write "+
+					"mappings folder: %w", err)
+			}
+		}
```

Comment:
I would raise this out of the loop:

```go
if len(mappings)>0 { ...
```

I think it makes the code read easier.